include/jemalloc/internal/hpdata.h

```diff
@@ -0,0 +1,123 @@
+#ifndef JEMALLOC_INTERNAL_HPDATA_H
+#define JEMALLOC_INTERNAL_HPDATA_H
+
+#include "jemalloc/internal/flat_bitmap.h"
+#include "jemalloc/internal/ph.h"
+#include "jemalloc/internal/ql.h"
+#include "jemalloc/internal/typed_list.h"
+
+/*
+ * The metadata representation we use for extents in hugepages.  While the PAC
+ * uses the edata_t to represent both active and inactive extents, the HP only
+ * uses the edata_t for active ones; instead, inactive extents are tracked by
+ * the hugepage they're in.
```

Comment:
` tracked by the hugepage they're in` is a bit unclear to me. That means they don't need metadata storage, other than their vaddr?

Reply:
Right; active vs. inactive state is tracked in the bitmap in the hpdata (and, in the purging stack commits, so is dirty state). I beefed up the comment some.